sdk/cosmos/azure-cosmos/azure/cosmos/aio/_container.py

```diff
@@ -1047,6 +1048,12 @@ async def execute_item_batch(
         :returns: A list representing the items after the batch operations went through.
         :raises ~azure.cosmos.exceptions.CosmosHttpResponseError: The batch failed to execute.
         :raises ~azure.cosmos.exceptions.CosmosBatchOperationError: A transactional batch operation failed in the batch.
+            CosmosBatchOperationError will have several attributes providing the user more information on the error:
+            - error_index: the index of the failed operation
+            - headers: the response headers for the failed operation
+            - status_code: the status code of the failed operation, since all other operations will fail with a 424
+            - message: the error message, also including the index of the failed operation
+            - operation_responses: the list of failed operation responses
```

Comment:
These shouldn't be documented here - they should be documented as attributes (`:ivar:` ) on the exception object itself.
Wouldn't hurt to throw in a code snippet into the class docstring for that exception object as well.

Reply:
Additionally they will all need to be properly typed. Just saying "the operation responses" isn't sufficient.

Reply:
we already had this for the class - does that mean I just need to change them to ivar instead?
<img width="565" alt="image" src="https://github.com/Azure/azure-sdk-for-python/assets/30335873/2911737f-0aba-49ea-b69b-97b19245fad0">

Reply:
Correct - documenting how to construct the error is of no customer value - as that's only ever used internally.
We should flesh out the class docstring to detail how the object should be used from a customer perspective.

And yes - you should be able to just tag the snippet within the relevant sample and reference it directly:
https://github.com/Azure/azure-sdk-for-python/blob/main/sdk/cosmos/azure-cosmos/azure/cosmos/container.py#L215-L223

Reply:
went ahead and updated both of these points, marking the attributes as `ivar` and adding the small sample snippet